### PR TITLE
Append API SPI page and try a better format

### PR DESCRIPTION
**Pull request by XdoctorwhoZ**

### CODE REVIEW

**Review by Wardru — APPROVED** [review]
Sounds good for me :)

File: api/spi_master.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+------------
+
++-----------------+-----+--------+-------------------------+------------------------+
+| Suffix          | QOS | Retain | Payload                 | Description            |
++=================+=====+========+=========================+========================+
+| info            | 0   | false  | :ref:`Payload Info`     | Interface heart beat   |
++-----------------+-----+--------+-------------------------+------------------------+
+| atts/data       | 0   | false  | :ref:`payload-spi-data` | Data recieved from spi |
++-----------------+-----+--------+-------------------------+------------------------+
+| cmds/data/read  | 0   | false  |                         |                        |
```
Comment (BCadet) [inline api/spi_master.rst:22]: no read or write cmds are required. only a "transfer" method is useful as the spi always read and write at the same time.  
So, when you want to write, you post you payload in the "transfer" cmd and you get a buffer (of the same size of pyaload) in the attd/data topic corresponding to the MISO data.

Comment (BCadet) [reply api/spi_master.rst:22]: When you want to read, you just put a payload of 0x00 in the transfer cmd and you get the MISO data line in the data atts

File: api/spi_slave.rst [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Spi Slave
+===============
+
+MQTT specifications for Panduza Spi Slave interfaces.
+
+.. code-block::
+
+    TYPE: "spi/slave"
```
Comment (BCadet) [inline api/spi_slave.rst:8]: For the slave, I think we should provide a "register map" (address/value) of the slave and a way for the MQTT to be notified on:
- a read access
- a write access

for each register.  

We also need a way to change the value of a register without unloading the register map

File: api/spi_master.rst (hunk 1 above)
Comment (XdoctorwhoZ) [reply api/spi_master.rst]: I updated spi master topics following this remark

File: api/spi_slave.rst (hunk 2 above)
Comment (XdoctorwhoZ) [reply api/spi_slave.rst:8]: We talk about it, memory map must be managed in an other way

**Review by BCadet — APPROVED** [review]

**Pull request merged**